x-pack/plugins/ml/common/util/record_utils.ts

```diff
@@ -0,0 +1,10 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+export const isRecord = <T = Record<string, any>>(arg: any): arg is T => {
```

Comment:
the name of this check might be confusing. it's not really checking that `arg` is a `Record` it's checking that `arg` is a populated object.

Reply:
Renamed here https://github.com/elastic/kibana/pull/90363/commits/bc435134b7a10eb60657c077bec7a52659473569